include/caffe/util/blocking_queue.hpp

```diff
@@ -0,0 +1,48 @@
+#ifndef CAFFE_UTIL_BLOCKING_QUEUE_H_
+#define CAFFE_UTIL_BLOCKING_QUEUE_H_
+
+#include <queue>
+#include <string>
+
+#include "caffe/common.hpp"
+
+namespace caffe {
+
+template<typename T>
+class blocking_queue {
```

Comment:
should be called `BlockingQueue` (https://google-styleguide.googlecode.com/svn/trunk/cppguide.html#Type_Names)